src/Common/src/Internal/Cryptography/CngCommon.SignVerify.cs

```diff
@@ -19,11 +21,23 @@ public static unsafe byte[] SignHash(this SafeNCryptKeyHandle keyHandle, ReadOnl
             byte[] signature = new byte[estimatedSize];
             int numBytesNeeded;
             ErrorCode errorCode = Interop.NCrypt.NCryptSignHash(keyHandle, pPaddingInfo, hash, hash.Length, signature, signature.Length, out numBytesNeeded, paddingMode);
+
+            if (errorCode == ErrorCode.STATUS_UNSUCCESSFUL)
+            {
+                errorCode = Interop.NCrypt.NCryptSignHash(keyHandle, pPaddingInfo, hash, hash.Length, signature, signature.Length, out numBytesNeeded, paddingMode);
+            }
+
             if (errorCode == ErrorCode.NTE_BUFFER_TOO_SMALL)
             {
                 signature = new byte[numBytesNeeded];
                 errorCode = Interop.NCrypt.NCryptSignHash(keyHandle, pPaddingInfo, hash, hash.Length, signature, signature.Length, out numBytesNeeded, paddingMode);
             }
+
+            if (errorCode == ErrorCode.STATUS_UNSUCCESSFUL)
```

Comment:
this might actually be third attempt, right? I'm fine either way

Reply:
It could be call 1: Unsuccessful, 2: too small, 3: Unsuccessful (large enough array), 4: last attempt, yeah.

But I'm pretty sure that this one uses the public keysize internally to understand too small, and all the failures were related to the private key.  So it's more likely (in retail): 1: Success, or 1: Unsuccessful, 2: last attempt, since we use a big enough buffer.